middleware/cors/cors_test.go

```diff
@@ -689,6 +689,43 @@ func Test_CORS_AllowCredentials(t *testing.T) {
 	}
 }
 
+// The Enhancement for issue #2804
+func Test_CORS_AllowPrivateNetworkAccess(t *testing.T) {
+	t.Parallel()
+
+	// Test scenario where AllowPrivateNetworkAccess is enabled
+	app := fiber.New()
+	app.Use(New(Config{
+		AllowPrivateNetworkAccess: true,
+	}))
+
+	handler := app.Handler()
+
+	ctx := &fasthttp.RequestCtx{}
+	ctx.Request.Header.SetMethod(fiber.MethodOptions)
+	ctx.Request.Header.Set(fiber.HeaderOrigin, "https://example.com")
+	ctx.Request.Header.Set("Access-Control-Request-Private-Network", "true")
```

Comment:
We are setting this here, we just need to check this header is in the request on line 237 of `cors.go`